src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -645,6 +649,11 @@ protected override TypeSymbol InferTypeOfVarVariable(DiagnosticBag diagnostics)
                         loopBinder.BindDeconstructionDeclaration(forStatement.Declaration, forStatement.Declaration, diagnostics);
                         break;
 
+                    case SyntaxKind.ForEachStatement:
+                        var foreachBinder = this.binder.GetBinder((ForEachStatementSyntax)statement);
+                        foreachBinder.BindForEachDeconstruction(diagnostics, foreachBinder);
```

Comment:
I'm not entirely sure, but Aleksey warned against taking such shortcuts in this [comment ](https://github.com/dotnet/roslyn/pull/12224#discussion_r68795192) on my last PR. 
In terms of binders, the above is staying as close as possible to what happens with `BindForEach` (before it calls into `BindForEachDeconstruction`).